processors/tracecompletiontime-deriver/src/main/java/org/hawkular/apm/processor/tracecompletiontime/TraceCompletionInformationInitiator.java

```diff
@@ -78,7 +77,7 @@ public TraceCompletionInformation processOneToOne(String tenantId,
                 ci.setCompletionTime(ct);
 
                 // Initialise any communications that need to be further processed
-                TraceCompletionInformationUtil.initialiseCommunications(ci, item.getNodes().get(0).getBaseTime(),
+                TraceCompletionInformationUtil.initialiseLinks(ci, item.getNodes().get(0).getBaseTime(),
```

Comment:
can be written as: `TraceCompletionInformationUtil.initialiseLinks(ci, n.getBaseTime(), 0, n)`

I think that signature of `TraceCompletionInformationUtil.initialiseLinks` can be refactored to `TraceCompletionInformation ci,  long baseDuration, Node n)`